app/Http/Controllers/ExportController.php

```diff
@@ -432,55 +511,47 @@ private function createCsv($rows, $headers) {
             fputcsv($csv, $row);
         }
         rewind($csv);
-        $csv = stream_get_contents($csv);
 
-        return $csv;
+        return stream_get_contents($csv);
     }
 
     /**
-     * Get a CSV or HTML of the export
+     * Get a CSV, JSON or HTML of the export
      *
-     * @var array  $csv      The data.
-     * @var string $title
-     * @var string $fileType 'csv' or 'html'
-     *
-     * @return The thing to present to the user.
+     * @param $csv
+     * @param string $title
+     * @param string $fileType
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Contracts\Routing\ResponseFactory|\Illuminate\Contracts\View\Factory|\Illuminate\Http\Response|\Illuminate\View\View
      */
-    private function getExport($csv, $title, $fileType) {
-        if ($fileType == self::CSV) {
+    private function getExport($csv, string $title, string $fileType) {
+        if ($fileType === self::CSV || $fileType === self::JSON) {
             return response($csv)
                 ->withHeaders([
-                    'Content-Type'        => 'text/csv',
+                    'Content-Type'        => "text/{$fileType}",
                     'Cache-Control'       => 'no-store, no-cache',
-                    'Content-Disposition' => 'attachment; filename="' . slug($title) . '.csv"',
+                    'Content-Disposition' => sprintf('attachment; filename="%s.%s"', slug($title), $fileType),
                 ]);
-        } else if ($fileType == self::JSON) {
-            return response($csv)
-                ->withHeaders([
-                    'Content-Type'        => 'text/json',
-                    'Cache-Control'       => 'no-store, no-cache',
-                    'Content-Disposition' => 'attachment; filename="' . slug($title) . '.json"',
-                ]);
-        } else {
-            return view('guild.export.generic', [
-                'data' => $csv,
-                'name' => $title,
-            ]);
         }
+
+        return view('guild.export.generic', [
+            'data' => $csv,
+            'name' => $title,
+        ]);

```

Comment:
I'm not going to hold up the PR over this or anything, but I'll probably put this back inside an `if` `else` when/if I'm working on it. Just FYI.